### PR TITLE
[receiver/collectd] Move to use HTTPServerSettings with collectdreceiver

### LINKED ISSUE #28811: Move collectdreceiver to move to use HTTPServerSettings and TimeoutSettings
### Component(s)

receiver/collectd

### Is your feature request related to a problem? Please describe.

The current approach to configuring collectdreceiver is rather old.

### Describe the solution you'd like

Move collectdreceiver to move to use HTTPServerSettings and TimeoutSettings

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by atoulme**
**Description:**
Overhauls collectdreceiver to use the latest config helper features

**Link to tracking Issue:**
Fixes #28811

**Documentation:**
No impact to docs. User interface remains the same.
Separate changelog to notice API breaking changes, as the Config struct is changing.

### CODE REVIEW

File: receiver/collectdreceiver/config_test.go [hunk 1]
```diff
@@ -53,7 +57,11 @@ func TestLoadConfig(t *testing.T) {
 			require.NoError(t, err)
 			require.NoError(t, component.UnmarshalConfig(sub, cfg))
 
-			assert.NoError(t, component.ValidateConfig(cfg))
+			if tt.wantErr == nil {
+				assert.NoError(t, component.ValidateConfig(cfg))
+			} else {
+				assert.Equal(t, tt.wantErr, component.ValidateConfig(cfg))
+			}
```
Comment (MovieStoreGuy) [inline receiver/collectdreceiver/config_test.go:64]: You should be able to use `assert.ErrorIs` here and avoid the if statement :)

**Review by MovieStoreGuy — APPROVED** [review]
Changes make sense to me, I don't see any potential breaking changes or changes in behaviour.

File: receiver/collectdreceiver/config_test.go (hunk 1 above)
Comment (atoulme) [reply receiver/collectdreceiver/config_test.go:64]: I tried just now, it didn't seem to work because the error on validation is built with `fmt.Errorf` and deep equality fails or something.

File: .chloggen/collectd-http-server-settings.yaml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+subtext:
+
+# If your change doesn't affect end users or the exported elements of any package,
+# you should instead start your pull request title with [chore] or use the "Skip Changelog" label.
+# Optional: The change log or logs in which this entry should be included.
+# e.g. '[user]' or '[user, api]'
+# Include 'user' if the change is relevant to end users.
+# Include 'api' if there is a change to a library API.
+# Default: '[user]'
+change_logs: []
```
Comment (dmitryax) [inline .chloggen/collectd-http-server-settings.yaml:27]: It's an enhancement for the end user but still a `breaking` change for Go API, which probably needs an additional changelog entry

Comment (atoulme) [reply .chloggen/collectd-http-server-settings.yaml:27]: Added, thanks

File: receiver/collectdreceiver/config.go [hunk 3]
```diff
@@ -4,16 +4,29 @@
...
-	"go.opentelemetry.io/collector/config/confignet"
+	"go.opentelemetry.io/collector/config/confighttp"
+	"go.opentelemetry.io/collector/exporter/exporterhelper"
 )
 
 // Config defines configuration for Collectd receiver.
 type Config struct {
-	confignet.TCPAddr `mapstructure:",squash"`
+	confighttp.HTTPServerSettings  `mapstructure:",squash"` // squash ensures fields are correctly decoded in embedded struct
+	exporterhelper.TimeoutSettings `mapstructure:",squash"` // squash ensures fields are correctly decoded in embedded struct
```
[the remark this answers is not in the document]
Comment (atoulme) [reply receiver/collectdreceiver/config.go]: OK, reverted this change.

File: .chloggen/collectd-http-server-settings-api.yaml [hunk 4]
```diff
@@ -0,0 +1,27 @@
+# Use this changelog template to create an entry for release notes.
+
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: breaking
+
+# The name of the component, or a single word describing the area of concern, (e.g. filelogreceiver)
+component: collectdreceiver
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: Move to use HTTPServerSettings and TimeoutSettings
```
Comment (dmitryax) [inline .chloggen/collectd-http-server-settings-api.yaml:10]:
```suggestion
note: Move to use confighttp.HTTPServerSettings
```

File: .chloggen/collectd-http-server-settings.yaml [hunk 5]
```diff
@@ -0,0 +1,27 @@
+# Use this changelog template to create an entry for release notes.
+
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: enhancement
+
+# The name of the component, or a single word describing the area of concern, (e.g. filelogreceiver)
+component: collectdreceiver
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: Move to use HTTPServerSettings and TimeoutSettings
```
Comment (dmitryax) [inline .chloggen/collectd-http-server-settings.yaml:10]:
```suggestion
note: Add support of confighttp.HTTPServerSettings
```

File: receiver/collectdreceiver/README.md [hunk 6]
```diff
@@ -32,7 +32,7 @@ and, `k` and `v` as the respective label values.
 
 The following settings are required:
 
-- `endpoint` (default = `localhost:8081`): Address to reach the desired Docker daemon.
+- `endpoint` (default = `localhost:8081`): Endpoint exposed by this receiver to send data.
```
Comment (dmitryax) [inline receiver/collectdreceiver/README.md:36]: Need to mention that other confighttp.HTTPServerSettings are supported as well

File: receiver/collectdreceiver/receiver.go [hunk 7]
```diff
@@ -28,60 +26,69 @@ var _ receiver.Metrics = (*collectdReceiver)(nil)
...
+	if err != nil {
+		return err
+	}
+	cdr.server.ReadTimeout = cdr.config.Timeout
+	cdr.server.WriteTimeout = cdr.config.Timeout
+	cdr.obsrecv, err = receiverhelper.NewObsReport(receiverhelper.ObsReportSettings{
+		ReceiverID:             cdr.createSettings.ID,
+		Transport:              "http",
+		ReceiverCreateSettings: cdr.createSettings,
+	})
```
Comment (dmitryax) [inline receiver/collectdreceiver/receiver.go:71]: Why is this moved to start?

**Review by dmitryax — APPROVED** [review]

File: receiver/collectdreceiver/receiver.go (hunk 7 above)
Comment (atoulme) [reply receiver/collectdreceiver/receiver.go:71]: there was something about having a clean receiver instantiation without too much fuss, and we cache createSettings as a field now with the new way to create the server, so it made sense to move all to Start.

File: receiver/collectdreceiver/README.md (hunk 6 above)
Comment (atoulme) [reply receiver/collectdreceiver/README.md:36]: Is there a good way to do that? looking at splunkhecreceiver, it's just as terse.

Comment (atoulme) [reply receiver/collectdreceiver/README.md:36]: ok I like what awsfirehosereceiver does:
> The configuration includes the Opentelemetry collector's server [confighttp](https://github.com/open-telemetry/opentelemetry-collector/tree/main/config/confighttp#server-configuration),
which allows for a variety of settings. Only the most relevant ones will be discussed here, but all are available.

File: receiver/collectdreceiver/receiver.go (hunk 7 above)
Comment (dmitryax) [reply receiver/collectdreceiver/receiver.go:71]: Ok, it just seems unrelated to the PR

**Review by dmitryax — APPROVED** [review]
